inputs/smart-list-input.vue

```diff
@@ -0,0 +1,172 @@
+<style lang="sass">
+  .smart-list-input {
+    align-items: center;
+    display: inline-flex;
+    flex: 1 0 auto;
+    height: 32px;
+    margin-bottom: 5px;
+    min-width: 100px;
+    position: relative;
+
+    .smart-list-add {
+      border-bottom: none;
+      flex: 1 0 auto;
+      width: auto;
+    }
+  }
+</style>
+
+<template>
+  <div class="smart-list-input" :class="classes">
+    <input
+      type="text"
+      class="ui-textbox__input smart-list-add"
+      ref="input"
+      :disabled="disabled"
+      v-model.trim="val"
+      @input="onChange"
+      @keydown.enter.prevent="onEnter"
+      @keydown.tab="addItem"
+      @keyup.comma="ignoreComma ? null : addItem()"
+      @keydown.delete="focusLastItem"
+      @keydown.left="focusLastItem"
+      @keydown.right="focusFirstItem"
+      @keydown.down="autocompleteFocus(false)"
+      @keydown.up.prevent="autocompleteFocus(true)"
+      @keypress="(event) => $emit('keypress', event)"
+      @focus="onFocus"
+      @blur="onBlur"
+      v-conditional-focus="focusOnInput" />
+    <autocomplete
+      v-if="showAutocomplete"
+      :args="autocomplete"
+      :query="val"
+      :disabled="disabled"
+      :focusIndex="autocompleteIndex"
+      :updateFocusIndex="updateFocusIndex"
+      :updateMatches="updateAutocompleteMatches"
+      :select="onAutocompleteSelect"
+      :unselect="unselect"
+      @resize="onResize">
+    </autocomplete>
+  </div>
+</template>
+
+<script>
+  import _ from 'lodash';
+  import autocomplete from './autocomplete.vue';
+
+  export default {
+    props: ['items', 'allowRepeatedItems', 'autocomplete', 'currentItem', 'disabled', 'isInitialFocus', 'ignoreComma'],
+    data() {
+      return {
+        val: '',
+        autocompleteIndex: null,
+        autocompleteOptions: [],
+        displayAutocomplete: true
+      };
+    },
+    computed: {
+      showAutocomplete() {
+        return this.autocomplete && this.displayAutocomplete;
+      },
+      focusOnInput() {
+        return _.isNull(this.currentItem);
+      },
+      classes() {
+        return [
+          { 'is-disabled': this.disabled }
+        ];
+      }
+    },
+    methods: {
+      onFocus() {
+        this.$emit('select', null);
+        this.$emit('focus');
+      },
+      onBlur() {
+        if (!this.showAutocomplete) {
+          // don't blur when we're displaying the autocomplete
+          this.$emit('blur');
+        }
+      },
+      onChange() {
+        if (!this.displayAutocomplete) {
```

Comment:
Can we replace the `if` with `this.displayAutocomplete = true`?

Reply:
It occurred to me that this might trigger a re-render, in that case the `if` is worthwhile.  In either case the assignment could be simplified to `= true`.

Reply:
This is all copied from `simple-list`